pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -46,8 +47,17 @@ var _ Interface = new(cadvisorClient)
 // The amount of time for which to keep stats in memory.
 const statsCacheDuration = 2 * time.Minute
 const maxHousekeepingInterval = 15 * time.Second
+const defaultHousekeepingInterval = 10 * time.Second
 const allowDynamicHousekeeping = true
 
+func init() {
+	// Override the default cAdvisor housekeeping interval.
+	if f := flag.Lookup("housekeeping_interval"); f != nil {
```

Comment:
Would it be OK to say that users should not change cAdvisor flags directly?
We do want to hide cAdvisor flags completely in the future right?

On Fri, Jan 8, 2016 at 1:13 PM, Tim St. Clair <notifications@github.com>
wrote:

> In pkg/kubelet/cadvisor/cadvisor_linux.go
> <https://github.com/kubernetes/kubernetes/pull/19374#discussion_r49238715>
> :
>
> >  const allowDynamicHousekeeping = true
> >
> > +func init() {
> > +	// Override the default cAdvisor housekeeping interval.
> > +	if f := flag.Lookup("housekeeping_interval"); f != nil {
>
> The problem with that is the way the code is currently set up there is no
> way to remove flags. If we add a kubelet flag, the cAdvisor flag will still
> be there (which is confusing). It would be a fairly large refactoring to
> consolidate all the cAdvisor flags so they're only added in the standalone
> version, and not the library version.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19374/files#r49238715>.
>